### DISCUSSION TITLE
Issue using 0.11.1: "Transient error [...] while exporting traces"

### DISCUSSION THREAD

**Discussion by rgilton** [dagger/dagger#7116] [Help] [answered]
Running a [simple script](https://gist.github.com/rgilton/37ee8f394ff36975f77031c1f1b9db0d) with 0.11.1 results in lots of error messages like this:

```
Transient error StatusCode.UNAVAILABLE encountered while exporting traces to localhost:4317, retrying in 1s.
Transient error StatusCode.UNAVAILABLE encountered while exporting traces to localhost:4317, retrying in 2s.
Transient error StatusCode.UNAVAILABLE encountered while exporting traces to localhost:4317, retrying in 4s.
Transient error StatusCode.UNAVAILABLE encountered while exporting traces to localhost:4317, retrying in 8s.
Transient error StatusCode.UNAVAILABLE encountered while exporting traces to localhost:4317, retrying in 16s.
Transient error StatusCode.UNAVAILABLE encountered while exporting traces to localhost:4317, retrying in 32s.
```

These error messages don't happen with 0.10.3.  Anyone know if this is a bug, or have I misconfigured something?

**Comment by wingyplus** [comment]
Did you run it with `dagger run` or just executing the script (or with `python`) directly?

**Reply by rgilton** [reply]
With python directly.

**Comment by helderco** [comment, marked as answer]
Sorry about that @rgilton! It's a bug. I thought `dagger run` was setting the telemetry env vars but it doesn't. I've actually been working on that this week so it should be fixed by next release. 

I'll see if there's an env var you can use to disable it in the meantime.

**Reply by rgilton** [reply]
No problem!  I'm actually quite relieved it's a bug, and not a misunderstanding on my part!

**Reply by helderco** [reply]
Try disabling with this (before `dagger.connection()`):
```python
import logging
from opentelemetry import trace

logging.getLogger("opentelemetry").addHandler(logging.NullHandler())
trace.get_tracer_provider().shutdown()
```

The logger statement is to avoid showing the "Already shutdown, dropping span" message.

**Reply by rgilton** [reply]
This seems to work, but I seem to get absolutely no output on the console any more (e.g. no build process info) -- is that expected?

**Reply by helderco** [reply]
No, that should disable only "opentelemetry" logs. What kind of output do you see without it? Can you share a bit more details on your setup? Do you having logging configured? Are you using the `log_output` config?

**Reply by rgilton** [reply]
The script I am running is in the gist that I linked to in the original question.

If I run that script with 0.10.3, then I get a bunch of console output, along these lines:

```
⠙ Downloading dagger CLI1: connect
⠋ Creating new Engine session1: starting engine 
⠇ Creating new Engine session1: starting engine [1.40s]
1: starting engine [0.06s]
1: starting session 
⠋ Creating new Engine session1: [2.28s] OK!
1: starting session [0.10s]
1: connect DONE

5: resolve image config for docker.io/library/alpine:3.18.6
5: > in from alpine:3.18.6
5: resolve image config for docker.io/library/alpine:3.18.6 DONE
[... etc ...]
```

If I run that script with 0.11.1 with the modification you have suggested, I get one message flashing up on the terminal (possibly "creating new engine session"), which clears pretty much immediately, and then there is zero output while all the work is done, leaving me back on my shell prompt with literally no lines of output from the script.:

```
(dagger-demo) [rob@zem dagger-demo(main)]$ ./go 
(dagger-demo) [rob@zem dagger-demo(main)]$ 
```



I don't know if this is a feature of 0.11.1, or something to do with the logging...

**Reply by johnstonematt** [reply]
I also have this issue with no output in 0.11

**Reply by helderco** [reply]
> The script I am running is in the gist that I linked to in the original question.

Quick tip: you don't need to pass the `client` around instance anymore, see the [Global Client](https://dagger.io/blog/simpler-reusability-with-the-global-client).

> I don't know if this is a feature of 0.11.1, or something to do with the logging...

It's from dagger 0.11, not Python specifically. You should see some progress while it's happening and in the end it goes away to leave only the script's own output.

If you run the script through `dagger run` you have more control. You can increase the verbosity with `-v` (up to `-vvv`):
```
dagger run -v python foo.py
```
Which will persist a small amount of that progress after the script finishes.

You can also return to the previous output by disabling the TUI:
```
dagger run --progress=plain python foo.py
```

**Reply by johnstonematt** [reply]
So, in 0.11, there is no way of getting the "complete" output without running `dagger run`?

If that is the case could I put in a huge request that that be reverted? The output from simply running `python foo.py` in 0.10 and 0.9 was incredibly helpful.

**Reply by rgilton** [reply]
> Quick tip: you don't need to pass the client around instance anymore, see the [Global Client](https://dagger.io/blog/simpler-reusability-with-the-global-client).

:smiley: 

> You should see some progress while it's happening and in the end it goes away to leave only the script's own output.

Yea, I saw literally that one line flash up and no other lines!

Ok, if I run it via `dagger run` in the way you say (didn't know I could do that!) then I do get more output.

**Reply by rgilton** [reply]
It was also useful to be able to run the script without having had to install the dagger binary separately: if it's just in a python script, it's just a `pip install dagger-io` to get it to work. Whereas if `dagger` has to be around then there has to be some way of installing that first...

**Reply by helderco** [reply]
> If that is the case could I put in a huge request that that be reverted?

Best create an issue asking for it. There was a similar issue here for `dagger run`:
- https://github.com/dagger/dagger/issues/7057

The difference is you need it on an SDK's _automatic provisioning_ (it's the SDK downloading and executing the CLI underneath).
